paasta_tools/list_marathon_service_instances.py

```diff
@@ -25,34 +25,114 @@
 
 - -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
 - -c <CLUSTER>, --cluster <CLUSTER>: Specify which cluster of services to read
+- -m, --minimal: Only show service instances that need bouncing
 """
 import argparse
 import sys
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
+from paasta_tools.marathon_tools import get_marathon_client
+from paasta_tools.marathon_tools import get_num_at_risk_tasks
+from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.utils import compose_job_id
+from paasta_tools.utils import decompose_job_id
 from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import use_requests_cache
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Lists marathon instances for a service.')
+    parser = argparse.ArgumentParser(
+        description='Lists marathon instances for a service.')
     parser.add_argument('-c', '--cluster', dest="cluster", metavar="CLUSTER",
                         default=None,
                         help="define a specific cluster to read from")
     parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
                         default=DEFAULT_SOA_DIR,
                         help="define a different soa config directory")
+    parser.add_argument('-m', '--minimal', dest='minimal', action='store_true',
+                        help="show only service instances that need bouncing")
     args = parser.parse_args()
     return args
 
 
+def long_job_id_to_short_job_id(long_job_id):
+    service, instance, _, __ = decompose_job_id(long_job_id)
+    return compose_job_id(service, instance)
+
+
+def get_current_apps():
+    marathon_config = load_marathon_config()
+    marathon_client = get_marathon_client(
```

Comment:
You are duplicating `list_all_marathon_app_ids` no?

Passing the client around only "feels" lame because you only need the client a single time in your script.
Please pass the client around, even if you only end up using it one time for `list_all_marathon_app_ids()`